stor/dx.py

```diff
@@ -95,32 +117,54 @@ def exists(self):
 
 
 class DXVirtualPath(DXPath):
+    """Class Handler for DXPath of form 'dx://project-{ID}:/a/b/c' or 'dx://a/b/c'"""
+
+    @property
+    def virtual_project(self):
+        return self.project
+
+    @property
+    def virtual_resource(self):
+        return self.resource
+
+    @property
+    def virtual_path(self):
+        return self
 
     @cached_property
     def canonical_project(self):
         """Returns the first project that matches the name that user has view access to.
         If no match is found, returns None
         """
-        # we need the DX project-ID to get the file_ID and full path
-        proj_dict = db.search.find_one_project(
-            name=self.project, level='VIEW', zero_ok=True, more_ok=False)
-        if proj_dict:
-            return proj_dict['id']
+        try:
+            if not dxb.verify_string_dxid(self.project, 'project'):
+                return self.project
+        except DXError:
+            try:
+                proj_dict = dxb.search.find_one_project(
```

Comment:
let's return different error messages when there is one project and multiple projects and tell the user which is which + return a useful exception type that you can catch, e.g.:

```
try:
    stor.Path('dx://...').describe()
except stor.dnanexus.InvalidProjectError:
    pass
```

some name that's clear when you read down code that uses it

and probably all of these need to inherit from `InvalidPath`???

and if there's zero, then that's essentially a `PathNotFound`, so maybe we want to have a hierarchy like this:

```
ProjectNotFoundError(NotFoundError)
```

```
DuplicateProjectError(DNAnexusError)
```